src/Caching/StackExchangeRedis/src/RedisCacheImpl.cs

```diff
@@ -0,0 +1,15 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.Caching.StackExchangeRedis;
+
+internal sealed class RedisCacheImpl : RedisCache
+{
+    public RedisCacheImpl(IOptions<RedisCacheOptions> optionsAccessor, ILogger<RedisCacheImpl> logger)
```

Comment:
This ends up being way cleaner than how we did it for `KestrelServerImpl` though. Leaving the implementation in `RedisCache` and just adding the public constructor here is nice.

I think we should add a `public RedisCacheImpl(IOptions<RedisCacheOptions> optionsAccessor)` constructor just in case someone out there was calling this on an `IServiceCollection` without logging. I think this would be cleaner than `.AddLogging()` in `.AddStackExchangeRedisCache().` If someone already has logging configured, it will light up. Otherwise, it won't add any unnecessary services.